File: 2677-cousins-in-binary-tree-ii/2677-cousins-in-binary-tree-ii.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public TreeNode ReplaceValueInTree(TreeNode root) {
        int sum1 = 0; 
        int sum2 =0;
        Queue<Tuple<TreeNode,int>> q = new Queue<Tuple<TreeNode,int>>();
        q.Enqueue(new Tuple<TreeNode,int>(root,0));
        for(int level = 1, sz = q.Count(); q.Count()>0; level++, sz = q.Count())
        {
            while(sz-- > 0)
            {
                Tuple<TreeNode,int>p = new Tuple<TreeNode,int>(q.Peek().Item1,q.Peek().Item2);
                q.Dequeue();
                if(level == 1)
                {
                    p.Item1.val = 0;
                    int x = p.Item1.right !=null ? p.Item1.right.val   :0;
                    int y = p.Item1.left!=null ? p.Item1.left.val  :0;
                    if(p.Item1.right != null)
                        {
                            q.Enqueue(new Tuple<TreeNode,int>(p.Item1.right,y));
                            sum2+=p.Item1.right.val;
                        }
                    if(p.Item1.left != null)
                        {
                            q.Enqueue(new Tuple<TreeNode,int>(p.Item1.left,x));
                            sum2+=p.Item1.left.val;
                        }
                }
                else
                {
                    p.Item1.val = sum1-p.Item1.val-p.Item2;
                    if(p.Item1.right != null)
                    {
                        int x = p.Item1.left != null ? p.Item1.left.val : 0;
                        q.Enqueue(new Tuple<TreeNode,int>(p.Item1.right,x));
                        sum2+=p.Item1.right.val;
                    }
                    if(p.Item1.left != null)
                        {
                            int x = p.Item1.right != null ? p.Item1.right.val : 0;
                            q.Enqueue(new Tuple<TreeNode,int>(p.Item1.left,x) );
                            sum2+=p.Item1.left.val;
                        }
                }
            }
            sum1= sum2;
            sum2=0;
        }return root;
        
    }
}

File: 2677-cousins-in-binary-tree-ii/2677-cousins-in-binary-tree-ii.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* replaceValueInTree(TreeNode* root) {
        int sum1 = 0; 
        int sum2 =0;
        queue<pair<TreeNode*,int>> q;
        q.push({root,0});
        for(int level = 1, sz = q.size(); !q.empty(); level++, sz = q.size())
        {
            while(sz--)
            {
                pair<TreeNode*,int>p =  q.front();
                q.pop();
                if(level == 1)
                {
                    p.first->val = 0;
                    if(p.first->right)
                        q.push({p.first->right,0});
                    if(p.first->left)
                        q.push({p.first->left,0});
                    continue;
                }
                else if(level == 2)
                {
                    p.first->val = 0; 
                    if(p.first->right)
                    {
                        int x = p.first->left ? p.first->left->val : 0;
                        q.push({p.first->right,x});
                        sum2+=p.first->right->val;
                    }
                    if(p.first->left)
                        {
                            int x = p.first->right ? p.first->right->val : 0;
                            q.push({p.first->left,x} );
                            sum2+=p.first->left->val;
                        }
                }
                else
                {
                    p.first->val = abs(sum1-p.first->val-p.second);
                    if(p.first->right)
                    {
                        int x = p.first->left ? p.first->left->val : 0;
                        q.push({p.first->right,x});
                        sum2+=p.first->right->val;
                    }
                    if(p.first->left)
                        {
                            int x = p.first->right ? p.first->right->val : 0;
                            q.push({p.first->left,x} );
                            sum2+=p.first->left->val;
                        }
                }
            }
            sum1= sum2;
            sum2=0;
        }return root;
    }
};